File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int minPrice = Integer.MAX_VALUE;
        int maxProfit = 0;
        int right = 0;
        while (right < prices.length) {
            if (prices[right] < minPrice) {
                minPrice = prices[right];
            } else if (maxProfit < prices[right] - minPrice) {
                maxProfit = prices[right] - minPrice;
            }
            right++;
        }
        
        return maxProfit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.kt
class Solution {
    fun maxProfit(prices: IntArray): Int {
        var minPrice = Int.MAX_VALUE
        var maxProfit = 0
        var right = 0
        
        while (right < prices.size) {
            if (minPrice > prices[right]) {
                minPrice = prices[right]
            } else if (maxProfit < prices[right] - minPrice) {
                maxProfit = prices[right] - minPrice
            }
            right++
        }
        
        return maxProfit
    }
}